pkg/controller/atlasproject/encryption_at_rest.go

```diff
@@ -140,57 +139,32 @@ func readSecretData(ctx context.Context, kubeClient client.Client, res common.Re
 	return result, nil
 }
 
-func createOrDeleteEncryptionAtRests(ctx *workflow.Context, projectID string, project *akov2.AtlasProject) workflow.Result {
-	encryptionAtRestsInAtlas, err := fetchEncryptionAtRests(ctx, projectID)
+func createOrDeleteEncryptionAtRests(ctx *workflow.Context, service encryptionatrest.EncryptionAtRestService, projectID string, encRest *encryptionatrest.EncryptionAtRest) workflow.Result {
+	encryptionAtRestsInAtlas, err := service.Get(ctx.Context, projectID)
 	if err != nil {
 		return workflow.Terminate(workflow.Internal, err.Error())
 	}
 
-	inSync, err := AtlasInSync(encryptionAtRestsInAtlas, project.Spec.EncryptionAtRest)
-	if err != nil {
-		return workflow.Terminate(workflow.Internal, err.Error())
-	}
+	inSync := encryptionatrest.EqualSpecs(encRest, encryptionAtRestsInAtlas)
 
 	if inSync {
 		return workflow.OK()
 	}
 
-	if err := syncEncryptionAtRestsInAtlas(ctx, projectID, project); err != nil {
-		return workflow.Terminate(workflow.Internal, err.Error())
-	}
-
-	return workflow.OK()
-}
-
-func fetchEncryptionAtRests(ctx *workflow.Context, projectID string) (*mongodbatlas.EncryptionAtRest, error) {
-	encryptionAtRestsInAtlas, _, err := ctx.Client.EncryptionsAtRest.Get(ctx.Context, projectID)
-	if err != nil {
-		return nil, err
-	}
-	ctx.Log.Debugf("Got EncryptionAtRests From Atlas: %v", *encryptionAtRestsInAtlas)
-	return encryptionAtRestsInAtlas, nil
-}
-
-func syncEncryptionAtRestsInAtlas(ctx *workflow.Context, projectID string, project *akov2.AtlasProject) error {
-	requestBody := mongodbatlas.EncryptionAtRest{
-		GroupID:        projectID,
-		AwsKms:         getAwsKMS(project),
-		AzureKeyVault:  getAzureKeyVault(project),
-		GoogleCloudKms: getGoogleCloudKms(project),
-	}
-
-	if err := normalizeAwsKms(ctx, projectID, &requestBody.AwsKms); err != nil {
-		return err
+	if encRest != nil {
+		if err := normalizeAwsKms(ctx, projectID, &encRest.AWS); err != nil {
+			return workflow.Terminate(workflow.Internal, err.Error())
+		}
```

Comment:
Why only AWS needs this normalization?

Reply:
This special case was implemented as part of https://github.com/mongodb/mongodb-atlas-kubernetes/pull/987. Maybe @igor-karpukhin has more historical context.